lib/logstash/codecs/cloudtrail.rb

```diff
@@ -20,17 +20,18 @@ def decode(data)
     decoded = LogStash::Json.load(@converter.convert(data))
     decoded['Records'].each do |event|
       event['@timestamp'] = event.delete('eventTime')
-
-      if event.has_key?("requestParameters")
-        if event['requestParameters'].has_key?("disableApiTermination")
-          if event['requestParameters']['disableApiTermination'].class != Hash
-            disableApiTermination = event['requestParameters'].delete('disableApiTermination')
-            event['requestParameters']['disableApiTermination']= {"value" => disableApiTermination}
+      if @event
+        if event.has_key?("requestParameters")
```

Comment:
This needs additional check to ensure `requestParameters` isn't null. Something like:
`event.has_key?("requestParameters") && ! event['requestParameters'].nil?`

Reply:
Needs to handle ```{"Records":[{"requestParameters":null}]}``` that is generated by cloudtrail. This fix sitll fails when requestParameter is set to null in a record and will crash the input for the pipeline